frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceForm.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { TextInput, FormGroup, Form } from '@patternfly/react-core';
+import { Identifier } from '~/types';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { UnitOption } from '~/utilities/valueUnits';
+import { validateDefaultCount, validateMinCount } from './utils';
+import CountFormField from './CountFormField';
+
+type NodeResourceFormProps = {
+  identifier: Identifier;
+  setIdentifier: UpdateObjectAtPropAndValue<Identifier>;
+  unitOptions?: UnitOption[];
+  existingIdentifier?: boolean;
+  isUniqueIdentifier?: boolean;
+};
+
+const NodeResourceForm: React.FC<NodeResourceFormProps> = ({
+  identifier,
+  setIdentifier,
+  unitOptions,
+  existingIdentifier,
+  isUniqueIdentifier,
+}) => {
+  const [validated, setValidated] = React.useState<'default' | 'error' | 'success'>('default');
+
+  React.useEffect(() => {
+    setValidated(isUniqueIdentifier ? 'default' : 'error');
+  }, [isUniqueIdentifier]);
+
+  return (
+    <Form>
+      <FormGroup label="Resource label" fieldId="resource-label">
+        <TextInput
+          value={identifier.displayName || ''}
+          onChange={(_, value) => setIdentifier('displayName', value)}
+        />
+      </FormGroup>
+
+      <FormGroup label="Resource identifier" fieldId="resource-identifier">
+        <TextInput
+          value={identifier.identifier || ''}
+          onChange={(_, value) => setIdentifier('identifier', value)}
+          isDisabled={
+            existingIdentifier &&
+            (identifier.identifier === 'cpu' || identifier.identifier === 'memory')
```

Comment:
I think this should be just:
```suggestion
     existingIdentifier?.identifier === 'cpu' || existingIdentifier?.identifier === 'memory'
```